client-sdk/ts-web/core/test/hdkey.test.ts

```diff
@@ -0,0 +1,46 @@
+import {HDKey} from '../src/hdkey';
```

Comment:
Can you add tests that will check if `getAccountSigner()` function call is passed:
- invalid mnemonic
- invalid account key number (i.e. bigger than `MaxAccountKeyNumber` in Go implementation).

Here are the corresponding tests in Go implementation:
https://github.com/oasisprotocol/oasis-core/pull/3918/files#diff-f92eefb38a2a59271dc7d0efad661a2cdb2034b2e1ba9df00672dc6a092b8b32R62-R81

BTW, in Go implementation the account key number has `uint32` type so there is no need to check if the number is < 0. How does the `number` type in TS behave? It will probably need a check that the passed number is not < 0.

Reply:
numbers are double precision floats. this one flows into a setUint32, which silently converts them to uint32 in some specific way

Reply:
@tjanez I've added a check for the account number

For the validity of the mnemonic this should be changed in the go implementation, the user implementation should raise a warning, but should not prevent the user from using any mnemonic, see [Bitcoinjs's reminder for developers](https://github.com/bitcoinjs/bip39#reminder-for-developers) : 

> Please remember to allow recovery from mnemonic phrases that have invalid checksums (or that you don't have the wordlist) When a checksum is invalid, warn the user that the phrase is not something generated by your app, and ask if they would like to use it anyway. This way, your app only needs to hold the wordlists for your supported languages, but you can recover phrases made by other apps in other languages.

Also from [BIP-0039](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#From_mnemonic_to_seed) : 

> Although using a mnemonic not generated by the algorithm described in "Generating the mnemonic" section is possible, this is not advised and software must compute a checksum for the mnemonic sentence using a wordlist and issue a warning if it is invalid.

Reply:
thanks for the explanation. I'm not opposed to adding the tests. a smoketest for different language mnemonic to make sure it works, and another non-official-test-vector test to make sure invalid account key number fails as expected